#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [t, aspectratio=169, allowframebreaks]
#+BEAMER_THEME: Amsterdam [compress]
#+latex_header:  \usepackage[numbers, authoryear, round]{natbib}
#+latex_header: \beamertemplatenavigationsymbolsempty
#+latex_header: \BeforeBeginEnvironment{frame}{\subsection{}}
#+latex_header: \usepackage[font=small,skip=0pt]{caption}
#+OPTIONS:  H:2 ^:nil

#+TITLE: Modelling the effects of domestication in Wheat through novel computer vision techniques
#+AUTHOR: Nathan Hughes


#+BEGIN_EXPORT latex
\addtobeamertemplate{block begin}{%
  \setlength{\textwidth}{1.0\textwidth}%
}{}

\addtobeamertemplate{block alerted begin}{%
  \setlength{\textwidth}{1.0\textwidth}%
}{}

\addtobeamertemplate{block example begin}{%
  \setlength{\textwidth}{1.0\textwidth}%
}{}


\AtBeginSection[]
  {
    \ifnum \value{framenumber}>3
      \begin{frame}<beamer>[noframenumbering]
      \frametitle{Outline}
      \tableofcontents[currentsection]
      \end{frame}
    \else
    \fi
  }

\setbeamertemplate{caption}[numbered]
\setbeamerfont{bibliography item}{size=\footnotesize}
\setbeamerfont{bibliography entry author}{size=\footnotesize}
\setbeamerfont{bibliography entry title}{size=\footnotesize}
\setbeamerfont{bibliography entry location}{size=\footnotesize}
\setbeamerfont{bibliography entry note}{size=\footnotesize}
\setbeamertemplate{bibliography item}{\insertbiblabel}
#+END_EXPORT


* Introduction & Background
** Introduction
*** What was the project?
The project aimed to use computational methods to answer biologically significant
 questions on wheat grain morphology and its changes throughout domestication (and by proxy polyploidization) using \micro-Computed-Tomography (CT) imaging techniques

*** Why is it important?
- The work carried out represents the first attempt to use \micro-CT imaging in order to investigate and model the effects of domestication and polyploidization in wheat
- Furthermore, it is unique in being one of the broadest studies of the genus /Triticum/

** Domestication/Polyploidization
*** What is Domestication?
- First suggested by Darwin, he described it as: "[the process] in which humans directly select for desirable traits" [fn:1: citet:darwin1868variation].
- It is estimated that the process of crop domestication began ~10,000 years ago [fn:2: citet:Charmet2011a]
- Over this time a large amount of variation in traits has appeared between the domesticated and the wild/primitive species
  - Barley, for example, has on average tripled its yield during domestication [fn:3: citet:Hebelstrup2017]

** Domestication/Polyploidization
\vspace{-0.5cm}
*** What is Polyploidization?                                         :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:
- This is the process by which organisms expand their ploidy count.
- /Ploidy/ - the number of sets of chromosomes an organism has.
- The full extent of the effects of polyploidy is an active area of research for many fields.

*** Figure                                                            :BMCOL:
   :PROPERTIES:
   :BEAMER_col: 0.5
   :END:
#+ATTR_LATEX: :width 4cm
#+CAPTION: Different Ploidy Counts Illustrated (en.wikipedia.org/wiki/Polyploid)
[[./ploidy.png]]


* Problems
** Problem 1
*** Lack of standards
Studies using \micro-CT are becoming more and more common
and yet:
- There is no standardisation of data handling [fn:5: citet:Tracy2017]
- No defined, or suggested methods for analysis [fn:6: citet:Hughes2017]
- Data is provided in various formats [fn:4: citet:cook2018barley]

This makes it difficult to collaborate, repeat,  or compare studies. It also adds confusion to the literature when each experiment, even within groups, is analysed differently.

** Problem 2
*** Big Data
The data generated by these experiments is huge! [fn:7: citet:Hughes2017]
- For a small/medium 3D image:
  - ~800MB
  - ~1000 images
- For an entire study of ~200 plants:
  - ~500GB
  - ~250,000 images

Extracting /meaningful/ data into manageable files is difficult


* Aims, Objectives & Goals
** Aims and Objectives
*** Aims
The main aim of this project has been to create software which is capable of providing answers and insight into the biological question proposed.

\vspace{0.3cm}

The software created should be:
- Reusable (between experiments)
- Modifiable (for when new data and parameters are introduced)
- Provide methods to common problems (and be free and accessible!)
- Allow quick and easy investigation of data


** Aims and Objectives
*** Objectives
In summary, the aims of this project are:
1. To provide a robust software library for analysing \micro-CT data
2. Create an easy to use GUI to automate a lot of the initial analysis
3. Provide evidence towards the domestication hypotheses [fn:8: discussed further in next slide]

** Research Goals
*** Hypotheses
1. Domesticated species differ from non-domesticated in terms of the traits examined
2. Increase in ploidy is positively associated with corresponding increases in the traits examined.

** Research Materials
*** Genotypes studied
1. /T. monococcum/ (2N|D) and /T. beoticum/ (2N|W) [fn:9: N denotes ploidy, W indicates wild and D domesticated]
2. /T. dicoccum/ (4N|D) and /T. dicoccoides/ (4N|W)
3. /T. spelta/ (6N|D) and /T. aestivum/ (6N|D)
4. /T. dicoccum/ (4N|D) and /T. durum/ (4N|D)
5. /T. beoticum/ (2N|W) and /T. dicoccoides/ (4N|W)
6. /T. dicoccum/ (4N|D) and /T. aestivum/ (6N|D)
7. /T. durum/ (4N|D) and /T. aestivum/ (6N|D)


** Research Materials
#+ATTR_LATEX: :width 8cm
#+NAME: fig:tree
#+CAPTION: Phylogentic Tree
[[./images/philotree.png]]



* Feature Extraction Methods
** Image Capture and Analysis
*** \micro-CT
- Imaging technologies are well-developed at the cellular and field scale
  - but there remains a gap in terms of extracting information about plant form and function at the whole organ level.
  - In particular, applying to large populations is difficult or expensive.
- \micro-CT imaging provides an method that captures structural information of plants without the need for destructive harvesting.
- Using this method, we can explore 3D traits in high-throughput experiments.

** Data Capture
*** CT Scanning software                                              :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:

The features/traits where extracted using an improved and optimised version of
our software which was used in our previous study.

\vspace{0.3cm}

Modifications were implemented to handle the wide range of diversity in the population of
this experiment

*** Image                                                             :BMCOL:
   :PROPERTIES:
   :BEAMER_col: 0.5
   :END:
   #+LATEX: \vspace{-0.3cm}
   #+ATTR_LATEX: :width 7cm
   #+CAPTION: Grains extracted from our imaging software and displayed in 3D
    [[./images/ctgrains.png]]

** Traits
*** Traits                                                            :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.45
   :END:
- Length
- Width
- Depth
- Volume
- Surface Area
- X,Y,Z coordinates of grains

*** Image							      :BMCOL:
   :PROPERTIES:
   :BEAMER_col: 0.6
   :END:
   #+ATTR_LATEX: :width 8cm
   #+CAPTION: Major features extracted from analysis
    [[./images/seeds.png]]


* Analysis Methods
** Why I don't trust the T-Test
*** Proof of deception
#+ATTR_LATEX: :width 8cm
#+NAME: fig:ttest
#+CAPTION: Exhibit A - Why the T-Test is evil
[[./images/ttest.png]]

** Bayesian Hypothesis Testing
*** Why?
    #+ATTR_BEAMER: :overlay +-
- "Despite their wide use in scientific journals ..., statistical  hypothesis tests add very little value to the products of research"  [fn:john: citet:Johnson1999a]
- It provides interpretable answers, such as “the true parameter \theta has a probability of 0.95 of falling in a 95% credible interval.”
- Allows for missing data points i.e. where a complete range of data is not possible

** Bayesian Model Used
*** Bayes states that
- $P(A|B) \propto P(B|A) \times P(A)$
  - The posterior is proportional to the likelihood times the prior
- $P(mean.1 | sample.1) \propto P(sample.1 | mean.1) \times  P(mean.1)$
*** Likelihood is described as
- $y_i^{(g)} \sim T(\nu, \mu, \sigma)$
  - $\nu$ (Degrees of freedom) is assumed similar for groups $g$
  - $\mu$ (mean) of groups is assumed the same
  - $\sigma$ (S.D.) is assumed the same

** Prior Mean $\mu$
*** Mean :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.6
   :END:
- $\mu_k \sim N(\bar{x},2s)$
  - The data are real-values and normal priors are applied (to ensure the posterior follows suit)
  - $2s$ - twice the S.D. ensures no values are favoured in the model

*** Distribution                                                      :BMCOL:
   :PROPERTIES:
   :BEAMER_col: 0.4
   :END:
\vspace{-0.8cm}
#+BEGIN_SRC python :results file :exports results
  import numpy as np
  import matplotlib.pyplot as plt
  filename='./images/mu.png'
  data = np.random.normal(1,10000,2000) # You are generating 1000 points between 0 and 1.
  count, bins, ignored = plt.hist(data,
				  20,
				  facecolor='green',
				  edgecolor='black',
				  alpha=0.5,
				  linewidth=1.2)

  plt.xlabel('Mean Value')
  plt.ylabel('Count')
  plt.title("Normal Distribution Histogram (Bin size 20)")
  # x_start, x_end, y_start, y_end
  plt.grid(True)
  plt.savefig(filename)
  return(filename)
#+END_SRC


#+ATTR_LATEX: :width 6.5cm
#+NAME: fig:mu
#+CAPTION: Normal$(\bar{x},2s)$
#+RESULTS:
[[file:./images/mu.png]]

** Prior Standard Deviations $\sigma$
*** Standard Deviations                                               :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.6
   :END:
- $\text{Uniform}(1,10000)$ is used
- Whilst no values in the model will have this range, it makes no difference due to random sampling
- Figure:[[fig:dist]] shows the distribution expected by random sampling
*** Distribution                                                      :BMCOL:
   :PROPERTIES:
   :BEAMER_col: 0.4
   :END:
\vspace{-0.8cm}
#+BEGIN_SRC python :results file :exports results
  import numpy as np
  import matplotlib.pyplot as plt
  filename='./images/dist.png'
  data = np.random.uniform(1,10000,2000) # You are generating 1000 points between 0 and 1.
  count, bins, ignored = plt.hist(data,
				  20,
				  facecolor='green',
				  edgecolor='black',
				  alpha=0.5,
				  linewidth=1.2)

  plt.xlabel('X~U[1,10000]')
  plt.ylabel('Count')
  plt.title("Uniform Distribution Histogram (Bin size 20)")
  # x_start, x_end, y_start, y_end
  plt.grid(True)
  plt.savefig(filename)
  return(filename)
#+END_SRC

#+ATTR_LATEX: :width 6.5cm
#+NAME: fig:dist
#+CAPTION: Uniform(1,10000)
#+RESULTS:
[[file:./images/dist.png]]

** Prior Degrees of freedom $\nu$
*** Degrees of freedom                                                :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.6
   :END:
- $\nu$ of 30 is used with an exponential distribution
- Shown in Figure:[[fig:expo]]


*** Distribution                                                      :BMCOL:
   :PROPERTIES:
   :BEAMER_col: 0.4
   :END:
\vspace{-0.8cm}
#+BEGIN_SRC python :results file :exports results
  import numpy as np
  import matplotlib.pyplot as plt
  filename='./images/dist2.png'
  data = np.random.exponential(30, 10000)
  count, bins, ignored = plt.hist(data,
				  40,
				  facecolor='green',
				  edgecolor='black',
				  alpha=0.5,
				  linewidth=1.2)

  plt.xlabel('Values')
  plt.ylabel('Count')
  plt.title("Exponential Distribution Histogram (Bin size 40)")
  plt.grid(True)
  plt.savefig(filename)
  return(filename)
#+END_SRC

#+ATTR_LATEX: :width 6.5cm
#+NAME: fig:expo
#+CAPTION: Exponential Distribution
#+RESULTS:
[[file:./images/dist2.png]]

** Sampling and Testing

*** Markov chain Monte Carlo
- 1000 random samples are drawn using Markov chain Monte Carlo [fn:krusche: citet:Kruschke2012]
  - This is done twice, independently to ensure convergence of randomness
- These provide a posterior of possibilities where the same mean could exist for the given data


* Analysis Example
** Example Input Data

#+ATTR_LATEX: :width 7.5cm
#+NAME: fig:bayesintro
#+CAPTION: Histogram of input data
[[file:exampledata.png]]

** Example Posterior

#+ATTR_LATEX: :width 7.5cm
#+NAME: fig:bayeshist
#+CAPTION: Histogram of posterior data
[[file:examplebayes.png]]

** Example Difference of Means

#+ATTR_LATEX: :width 7.5cm
#+NAME: fig:bayeshist2
#+CAPTION: Histogram of posterior data subtracted
[[file:examplebayes2.png]]

** Example Forest Plot
#+ATTR_LATEX: :width 7.5cm
#+NAME: fig:bayeshist3
#+CAPTION: Forest Plot of both chains (bold is 95% of data)
[[file:examplebayes3.png]]

* Results

** Einkorn Wild (2N|W) and Einkorn Domesticate (2N|D) ($P<0.01$)

*** Boxplots                                                          :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:
#+ATTR_LATEX: :width 4.5cm
#+NAME: fig:mu
#+CAPTION: Boxplot for volume
[[file:./images/monobox.png]]

*** Difference of means                                               :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:

#+ATTR_LATEX: :width 4.5cm
#+NAME: fig:mu
#+CAPTION: Difference of means
[[file:./images/monodiff.png]]


** Emmer Wild (4N|W) and Emmer Domesticated (4N|D) ($P = 0.032$)
*** Boxplots                                                          :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:
#+ATTR_LATEX: :width 4.5cm
#+NAME: fig:mu
#+CAPTION: Boxplot for width
[[file:./images/dicbox.png]]

*** Difference of means                                               :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:

#+ATTR_LATEX: :width 4.5cm
#+NAME: fig:mu
#+CAPTION: Difference of means
[[file:./images/dicdiff.png]]



** Spelt (6N|D) and Bread wheat (6N|D) ($P = 0.11$)

*** Boxplots                                                          :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:

#+ATTR_LATEX: :width 4.5cm
#+NAME: fig:mu
#+CAPTION: Boxplot for width
[[file:./images/speltbox.png]]

*** Difference of means                                               :BMCOL:
   :PROPERTIES:
   :BEAMER_env: block
   :BEAMER_col: 0.5
   :END:

#+ATTR_LATEX: :width 4.5cm
#+NAME: fig:mu
#+CAPTION: Difference of means
[[file:./images/speltdiff.png]]






* Software
** Live Demo!

*** CT Analysing Library
- The Library written in Python which controls all the analysis and nitty-gritty organisation

*** CT Analysing GUI
- A click based interface designed for rapid analysis and investigation of experiments
- Uses CT Analysing Library as its backend!

** References
:PROPERTIES:
:BEAMER_opt: allowframebreaks,label=
:END:

bibliography:library.bib
bibliographystyle:unsrtnat
